aiida_testing/export_cache/_fixtures.py

```diff
@@ -0,0 +1,327 @@
+                if entry.link_label not in self._hash_ignored_inputs
+            }
+        ]
+        return objects
+
+    monkeypatch.setattr(Code, "_get_objects_to_hash", mock_objects_to_hash_code)
+    monkeypatch.setattr(CalcJobNode, "_get_objects_to_hash", mock_objects_to_hash_calcjob)
+
+
+@pytest.fixture(scope='function')
+def run_with_cache(export_cache, load_cache):
+    """
+    Fixture to automatically import an aiida graph for a given process builder.
+    """
+    def _run_with_cache(
+        builder: ty.Union[dict, ProcessBuilder
+                          ],  #aiida process builder class, or dict, if process class is given
+        process_class=None,
+        label: str = '',
+        data_dir: ty.Union[str, pathlib.Path] = 'data_dir',
+    ):
+        """
+        Function, which checks if a aiida export for a given Process builder exists,
+        if it does it imports the aiida graph and runs the builder with caching.
+        If the cache does not exists, it still runs the builder but creates an
+        export afterwards.
+
+        Inputs:
+
+        builder : AiiDA Process builder class,
+        data_dir: optional
+            Absolute path of the directory where the exported workchain graphs are
+            stored.
+
+        ignore_nodes : list string, ignore input nodes with these labels/link labels to ignore in hash.
+        # needed?
+        """
+
+        cache_exists = False
+        if data_dir is 'data_dir':
+            cwd = pathlib.Path(os.getcwd())  # Might be not the best idea.
+            data_dir = (cwd / 'data_dir')  # TODO: get from config?
+
+        bui_hash, input_nodes = get_hash_process(builder)
+
+        if process_class is None:
+            process_class = builder.process_class
+        name = label + str(process_class).split('.')[-1].strip("'>") + '-nodes-' + bui_hash
+        print(name)
+
+        # check existence
+        full_import_path = str(data_dir) + '/' + name + '.tar.gz'
+        print(full_import_path)
+        cache_path = pathlib.Path(full_import_path)
+        if cache_path.exists():
+            cache_exists = True
+
+        if cache_exists:
+            # import data from previous run to use caching
+            load_cache(path_to_cache=full_import_path)
+
+        # now run process/workchain whatever
+        with enable_caching():  # should enable caching globally in this python interpreter
+            #yield #test is running
+            #res, resnode = run_get_node(builder)
+            res, resnode = run_get_node(process_class, **builder)
+
+        # This is executed after the test
+        if not cache_exists:
+            # TODO create datadir if not existent
+
+            # in case of yield:
+            # is the db already cleaned?
+            # since we do not the stored process node we try to get it from the inputs,
+            # i.e to which node they are all connected, with the lowest common pk
+            union_pk = ty.Set()
```

Comment:
This should probably be ``union_pk : ty.Set[str] = set()``. The ``typing`` classes are usually not for instantiating, but for the type annotations only.

Here the problem is that ``mypy`` cannot figure out which type the _content_ of the set has, because it is empty. So it needs help, which the ``ty.Set[str]`` provides.